File: ruby/src/select-1.rb
#!/usr/bin/env ruby
# -*- coding: utf-8 -*-
#
require 'sparql/client'

ENDPOINT = "http://dbpedia.org/sparql"

sparql = SPARQL::Client.new(ENDPOINT)
result = sparql.select.where([:s, :p, :o]).limit(10)

result.each_solution do |solution|
  puts solution.inspect
end

File: js/src/select1.js
const SparqlClient = require('sparql-client-2');
const SPARQL = SparqlClient.SPARQL;
const endpoint = 'http://ja.dbpedia.org/sparql';
const query = 
  SPARQL`SELECT *
         WHERE {
           ?s ?p ?o .
         }
         LIMIT 10`;
const client = new SparqlClient(endpoint)
  .register({
    dbp: 'http://ja.dbpedia.org/resource/',
    dbo: 'http://dbpedia.org/ontology/'
  });
 
client.query(query)
  .execute()
  .then(function (results) {
    console.dir(results, {depth: null});
  })
  .catch(function (error) {
    console.log("Error: " + error);
  });
